grails-app/services/org/pih/warehouse/invoice/PrepaymentInvoiceService.groovy

```diff
@@ -13,6 +13,40 @@ class PrepaymentInvoiceService {
 
     InvoiceService invoiceService
 
+    Invoice generatePrepaymentInvoice(Order order) {
+        if (order.orderItems.any { it.hasInvoices } || order.orderAdjustments.any { it.hasInvoices }) {
+            throw new Exception("Some order items or order adjustments for this order already have been invoiced")
+        }
+
+        Invoice invoice = invoiceService.createFromOrder(order)
+        invoice.invoiceType = InvoiceType.findByCode(InvoiceTypeCode.PREPAYMENT_INVOICE)
+        invoiceService.createOrUpdateVendorInvoiceNumber(invoice, order.orderNumber + Constants.PREPAYMENT_INVOICE_SUFFIX)
+
+        if (order?.orderItems?.empty && order?.orderAdjustments?.empty) {
+            throw new Exception("No order items or order adjustments found for given order")
+        }
+
+        BigDecimal prepaymentPercent = order.paymentTerm?.prepaymentPercent?:100
+
+        order.activeOrderItems.each { OrderItem orderItem ->
+            InvoiceItem invoiceItem = createFromOrderItem(orderItem)
+            // This is prepayment item, we need to adjust the amount according to the payment terms
+            invoiceItem.amount = invoiceItem.amount * (prepaymentPercent / 100)
```

Comment:
didn't you say amount wasn't being used?

Reply:
This `amount` is the one that is going to be used from now (this is a field on invoice item). The `totalAmount` transient was the deprecated one.